onnxscript/rewriter/rules/fusion/_rotary_embedding.py

```diff
@@ -43,20 +43,10 @@ def pattern(self, op, x, freqs, start1, end1, start2, end2, one1, one2):
     def check(self, op, x, start1, end1, start2, end2, one1, one2, **_) -> pattern.MatchResult:  # type: ignore[name-defined]
         check_result = pattern.MatchResult()
 
-        def is_one(val):
-            """Check if val is a 0/1 dimensional tensor with a single element equal to 1."""
-            np_val = _ir_utils.get_numpy_value(val)
-            return (
-                np_val is not None
-                and np_val.size == 1
-                and np_val.ndim <= 1
-                and np_val.item() == 1
-            )
-
-        if not is_one(one1):
-            return check_result.fail("Unsqueeze axes is not [1]", one1)
-        if not is_one(one2):
-            return check_result.fail("Unsqueeze axes is not [1]", one2)
+        if not _ir_utils.is_singleton_value(one1, 1, rank=(0, 1)):
+            return check_result.fail("Unsqueeze axes is not [1] or 1", one1)
+        if not _ir_utils.is_singleton_value(one2, 1, rank=(0, 1)):
+            return check_result.fail("Unsqueeze axes is not [1] or 1", one2)
```

Comment:
Unsqueeze must take a 1d axes. I think the previous pattern was due to an incorrect logic from torchlib which was fixed later